auth/user_mgt.go

```diff
@@ -498,6 +499,47 @@ func (c *baseClient) GetUserByPhoneNumber(ctx context.Context, phone string) (*U
 	})
 }
 
+// GetUserByProviderID gets the user data for the user corresponding to a given provider id.
```

Comment:
Unless this is a literal, capped "ID" would probably be preferred.

Reply:
Done.